hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/EncryptedFileSystem.java

```diff
@@ -98,6 +100,23 @@ public boolean rename(Path src, Path dst) throws IOException {
         return renamed;
     }
 
+    @Override
+    public boolean delete(Path path, boolean recursive) throws IOException {
+        FileStatus status;
+        try {
+            status = getFileStatus(path);
+        } catch (FileNotFoundException e) {
+            return false;
+        }
+
+        //Delete the key material if the path is a file
+        if (status.isFile()) {
+            keyStore.remove(path.toString());
```

Comment:
Both key storage strategies swallow failures, assuming that the underlying FS returns false in the case of a nonexistent file, which seems to be the case at least for hdfs/s3.